userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -1667,13 +1683,64 @@ static int32_t unsupported_config(struct scap_engine_handle engine, const char*
 	return SCAP_FAILURE;
 }
 
+static int32_t scap_bpf_handle_tp_mask(struct scap_engine_handle engine, uint32_t op, uint32_t tp)
+{
+	struct bpf_engine *handle = engine.m_handle;
+
+	int prg_idx = -1;
+	for (int i = 0; i < handle->m_bpf_prog_cnt; i++)
+	{
+		const tp_values val = tp_from_name(handle->m_bpf_progs[i].name);
+		if (val == tp)
+		{
+			prg_idx = i;
+			break;
+		}
+	}
+
+	// We want to unload a never loaded tracepoint
+	if (prg_idx == -1 && op != SCAP_TPMASK_SET)
+	{
+		return SCAP_SUCCESS;
+	}
+	// We want to load an already loaded tracepoint
+	if (prg_idx >= 0 && op != SCAP_TPMASK_UNSET)
+	{
+		return SCAP_SUCCESS;
+	}
+
+	if (op == SCAP_TPMASK_UNSET)
```

Comment:
If we are unsetting a tracepoint, we close its fd and event_fd, decrease the `m_bpf_prog_cnt` counter, and memmove all the data whose idx is `> m_bpf_prog_cnt` to the left.